refactor (#184036117) refactor basicbutton to remove web …

nullcitizen opened the pull request:
…component dependency

copies css from web-components into react project. copies dom tree from web components into render.
removes component import

[Ticket TICKET_ID](https://www.pivotaltracker.com/story/show/184036117)

nullcitizen committed:
refactor(atlas-react): #184036117 refactor basicbutton to remove web component dependency

copies css from web-components into react project. copies dom tree from web components into
render.
removes component import

nullcitizen commented on packages/atlas-react/src/components/BasicButton/BasicButton.scss:
```diff
@@ -1,7 +1,159 @@
+  --#{$namespace}-button-align: center;
+}
+
+.BasicButton {
```
there's some cleanup that could be done in this file but I'm really hoping for the chance to revert this in January so I'm not worried about it right now

demogar replied:
sounds good.

demogar commented on packages/atlas-react/src/components/BasicButton/BasicButton.scss:
```diff
@@ -1,7 +1,159 @@
+  }
+
+  .trailingIcon {
+    i {
```
what happens if we just use `←`, for example?

nullcitizen replied:
ah, perhaps I should have just converted this to .trailingIcon > *

nullcitizen committed:
refactor(atlas-react): #184036117 refactor css to apply trailing icon style to any child

rather than just i tags, trailingicons styles should apply to anything put in there
